WooCommerce/Classes/Extensions/String+Helpers.swift

```diff
@@ -68,4 +68,150 @@ extension String {
     var isNotEmpty: Bool {
         return !isEmpty
     }
+
+    /// Compares two strings as versions.
+    ///
+    /// Returns `orderedAscending` if the argument represents a newer version
+    /// Returns `orderedSame` if the argument represents the same version
+    /// Returns `orderedDescending` if the argument represents an older version
+    ///
+    func compareAsVersion(to: String) -> ComparisonResult {
+
+        /// Replace _ - and + with a .
+        ///
+        func replaceUnderscoreDashAndPlusWithDot(inString: String) -> String {
+            var outString = inString
+            let replaceTargets = ["_", "-", "+"]
+            for replaceTarget in replaceTargets {
+                outString = outString.replacingOccurrences(of: replaceTarget, with: ".")
+            }
+            return outString
+        }
+
+        /// Insert a . before and after any non number
+        ///
+        func insertDotsBeforeAndAfterAnyNonNumber(inString: String) -> String {
+            guard inString.count > 1 else {
+                return inString
+            }
+
+            /// Initialize our output with the first character of the input string
+            var outString = String(inString[inString.index(inString.startIndex, offsetBy: 0)])
+
+            /// Loop over the remaining characters in the string, inserting . as needed
+            for index in 1...inString.count - 1 {
+                let characterAtIndex = inString[inString.index(inString.startIndex, offsetBy: index)]
+                let characterBefore = inString[inString.index(inString.startIndex, offsetBy: index - 1)]
+
+                let characterAtIndexIsNumberOrDot = characterAtIndex.isNumber || characterAtIndex == "."
+                let characterBeforeIsNumberOrDot = characterBefore.isNumber || characterBefore == "."
+
+                if !characterAtIndexIsNumberOrDot && characterBefore.isNumber {
+                    outString += "."
+                }
+
+                if characterAtIndex.isNumber && !characterBeforeIsNumberOrDot {
+                    outString += "."
+                }
+
+                outString += String(characterAtIndex)
+            }
+
+            return outString
+        }
+
+        /// Ranked per https://www.php.net/manual/en/function.version-compare.php
+        ///
+        enum ComponentScore: Comparable {
```

Comment:
nit: this enum could be private within the struct if you're moving the logic to a separate comparator struct